File: src/main/java/ro/mpp2024/Domain/Pair.java
package ro.mpp2024.Domain;

import java.util.Objects;

public class Pair<Id1,Id2> {
    private Id1 id1;
    private Id2 id2;

    public Pair(Id1 id1, Id2 id2) {
        this.id1 = id1;
        this.id2 = id2;
    }

    public Id1 getId1() {
        return id1;
    }

    public void setId1(Id1 id1) {
        this.id1 = id1;
    }

    public Id2 getId2() {
        return id2;
    }

    public void setId2(Id2 id2) {
        this.id2 = id2;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Pair<?, ?> pair = (Pair<?, ?>) o;
        return Objects.equals(id1, pair.id1) && Objects.equals(id2, pair.id2);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id1, id2);
    }

    @Override
    public String toString() {
        return "Pair{" +
                "id1=" + id1 +
                ", id2=" + id2 +
                '}';
    }
}

File: SwimmingCompetitionCSharpGUI/Domain/Pair.cs
namespace SwimmingCompetition.Domain;

public class Pair<TId1, TId2>
{
    public TId1 Id1 { get; set; }
    public TId2 Id2 { get; set; }

    public Pair(TId1 id1, TId2 id2)
    {
        Id1 = id1;
        Id2 = id2;
    }

    protected bool Equals(Pair<TId1, TId2> other)
    {
        return EqualityComparer<TId1>.Default.Equals(Id1, other.Id1) && EqualityComparer<TId2>.Default.Equals(Id2, other.Id2);
    }

    public override bool Equals(object? obj)
    {
        if (ReferenceEquals(null, obj)) return false;
        if (ReferenceEquals(this, obj)) return true;
        if (obj.GetType() != this.GetType()) return false;
        return Equals((Pair<TId1, TId2>)obj);
    }

    public override int GetHashCode()
    {
        return HashCode.Combine(Id1, Id2);
    }

    public override string ToString()
    {
        return $"{nameof(Id1)}: {Id1}, {nameof(Id2)}: {Id2}";
    }
}